### DISCUSSION TITLE
Store videos/audios locally to listen from FreeTube

### DISCUSSION THREAD

**Discussion by LuisHPorras** [FreeTubeApp/FreeTube#2366] [Q&A] [answered]
Hi, Im new using FreeTube and I wonder if there is a way to store locally the videos/audios so the program doesnt have to download it everytime I listen to a song or similar. I know you can download them, but Im thinking about having the FreeTube experience as it is right now but being able to check which videos to store locally.
Thanks in advance!

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi @LuisHPorras im not sure what your question is. What i make out from this is that u know how to download videos and your question is whether u can play those videos in FT. If i misunderstood your question could u clarify your question please?

**Comment by LuisHPorras** [comment, marked as answer]
Yep more or less that is my question. But not using FT as a simpler media player but keeping the UX as closest to Youtube as possible, hypothetical situation:

I open FT and I search for xxxxxx song, which is one I really like and usually listen. The browser shows me a list of results, I pick the one I'm used to. Now the song (video or audio) is playing from FT. 

The question is, is it posible to follow that same scenario but with the song previously being saved in my local storage so it doesn't have to be downloaded again?

If I'm not being clear enough ask me specific questions, I'll be glad to answer. Thanks!

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
> The question is, is it posible to follow that same scenario but with the song previously being saved in my local storage so it doesn't have to be downloaded again?

So u have downloaded the video, then follow the same scenario and u wish that the video plays without doing any buffering because u already downloaded it?

**Reply by LuisHPorras** [reply]
Yep that should be it.

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
Thats unfortunately not a thing in FT at the moment and i honestly dont know if that is even possible.

@absidue thoughts?

**Reply by absidue** [reply]
Adding something like that is definitely a possibility. Caching all videos is going to eat up a lot of storage space on your computer, therefore I don't think it's something we should consider implementing before playlists are fully implemented. We could have a special playlist that caches videos once you add a video to that playlist, it could then download that video in the quality and format that you selected in the settings and use the cached version for future playback unless you change the quality or format, at which point it will need to rely on streaming it from the internet again.

@LuisHPorras Please [create a feature request](https://github.com/FreeTubeApp/FreeTube/issues/new?assignees=&labels=enhancement&template=feature_request.yaml&title=%5BFeature+Request%5D%3A+) for it so that people can find it again in the future if anyone decides to implement it.

**Reply by LuisHPorras** [reply]
Done, here is the [Issue](https://github.com/FreeTubeApp/FreeTube/issues/2374). Thank you both @absidue @efb4f5ff-1298-471a-8973-3d47447115dc
